broker-daemon/broker-rpc/broker-rpc-server.js

```diff
@@ -104,10 +78,41 @@ class BrokerRPCServer {
    * @returns {void}
    */
   listen (host) {
-    const rpcCredentials = createCredentials.apply(this)
+    const rpcCredentials = this.createCredentials()
     this.server.bind(host, rpcCredentials)
     this.server.start()
   }
+
+  /**
+   * Creates grpc server credentials for the broker rpc server
+   *
+   * @return {grpc.Credentials}
+   */
+  createCredentials () {
+    if (IS_PRODUCTION && this.disableAuth) {
+      throw new Error(`Cannot disable SSL in production. Set DISABLE_AUTH to FALSE.`)
+    }
+
+    if (this.disableAuth) {
+      this.logger.warn('DISABLE_AUTH is set to TRUE. Connections to the broker will be unencrypted. This is suitable only in development.')
+      return grpc.ServerCredentials.createInsecure()
+    }
+
+    const key = readFileSync(this.privKeyPath)
+    const cert = readFileSync(this.pubKeyPath)
+
+    this.logger.debug(`Securing gRPC connections with SSL: key: ${this.privKeyPath}, cert: ${this.pubKeyPath}`)
+
+    return grpc.ServerCredentials.createSsl(
+      null, // no root cert needed for server credentials
+      [{
+        private_key: key,
+        // TODO: build a real cert chain instead of just using the root cert
```

Comment:
do we ever expect to build a cert chain for the broker?

Reply:
This was copy pasta. I dont think we expect the broker to need anything more than pub/private for now